notebook/01_tutorial.ipynb

```diff
@@ -47,207 +69,159 @@
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 3,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "class VolumeFilterNode(Node):\n",
-      "\n",
-      "    def columns_setup(self):\n",
-      "        self.required = {\"asset\": \"int64\",\n",
-      "                         \"volume\": \"float64\"}\n",
-      "        self.addition = {\"mean_volume\": \"float64\"}\n",
-      "\n",
-      "    def process(self, inputs):\n",
-      "        \"\"\"\n",
-      "        filter the dataframe based on the min and max values of the average\n",
-      "        volume for each fo the assets.\n",
-      "\n",
-      "        Arguments\n",
-      "        -------\n",
-      "         inputs: list\n",
-      "            list of input dataframes.\n",
-      "        Returns\n",
-      "        -------\n",
-      "        dataframe\n",
-      "        \"\"\"\n",
-      "\n",
-      "        input_df = inputs[0]\n",
-      "        volume_df = input_df[['volume', \"asset\"]].groupby(\n",
-      "                [\"asset\"]).mean().reset_index()\n",
-      "        volume_df.columns = [\"asset\", 'mean_volume']\n",
-      "        merged = input_df.merge(volume_df, on=\"asset\", how='left')\n",
-      "        if 'min' in self.conf:\n",
-      "            minVolume = self.conf['min']\n",
-      "            merged = merged.query('mean_volume >= %f' % (minVolume))\n",
-      "        if 'max' in self.conf:\n",
-      "            maxVolume = self.conf['max']\n",
-      "            merged = merged.query('mean_volume <= %f' % (maxVolume))\n",
-      "        return merged\n",
-      "\n"
-     ]
-    }
-   ],
-   "source": [
-    "import inspect\n",
-    "from gquant.plugin_nodes.transform import VolumeFilterNode\n",
+    "In Python, a gQuant task-spec is defined as a dictionary with the following fields:\n",
+    "- `id`\n",
+    "- `type`\n",
+    "- `conf`\n",
+    "- `inputs`\n",
+    "- `filepath`\n",
     "\n",
-    "print(inspect.getsource(VolumeFilterNode))"
+    "As a best practice, we recommend using the `TaskSpecSchema` class for these fields, instead of strings.\n",
+    "\n",
+    "The `id` for a given task must be unique within a task graph. To use the result(s) of other task(s) as input(s) of a different task, we use the id(s) of the former task(s) in the `inputs` field of the next task.\n",
+    "\n",
+    "The `type` field contains the class of the compute task or **plugin**. gQuant includes several example plugins. These can be found in `gquant.plugin_nodes`. Click [here](#plugin_example) to see a gQuant plugin example.\n",
```

Comment:
this `plugin` really confuses me. @avolkov1, do you have a better name for it?

Reply:
Yea, node types or maybe node class makes sense. Plugin was meant to inform that one can create their own nodes. Maybe operator class.

""The `type` field contains the class of the compute task or node class. gQuant includes a collection of node classes. These can be found in `gquant.plugin_nodes`. Click [here](#plugin_example) to see a gQuant node class example."


""The `type` field contains the class of the compute task or operator class. gQuant includes a collection of operators. These can be found in `gquant.plugin_nodes`. Click [here](#plugin_example) to see a gQuant operator example."

Then we should rename plugins to operators maybe.

Reply:
I like this one, but I support your decision:

"The type field contains the node type to use for the compute task. gQuant includes a collection of node classes. These can be found in gquant.plugin_nodes. Click here to see a gQuant node class example."

Finally, could you please copy and paste the 'here' link url address? The url in the comment is wrong.

Thanks!